q2cli/builtin/tools.py

```diff
@@ -511,72 +511,141 @@ def _merge_metadata(paths):
     return metadata
 
 
-@tools.command(short_help='View a QIIME 2 Visualization.',
-               help="Displays a QIIME 2 Visualization until the command "
-                    "exits. To open a QIIME 2 Visualization so it can be "
-                    "used after the command exits, use 'qiime tools extract'.",
+def get_free_port():
+    """Finds an available port on the system.
+    """
+    import socket
+
+    with socket.socket() as _socket:
+        _socket.bind(('localhost', 0))
+        return _socket.getsockname()[1]
+
+
+@tools.command(short_help='View a QIIME 2 Result.',
+               help="Displays a QIIME 2 Result until the command exits. To "
+                    "open a QIIME 2 Visualization so it can be used after the "
+                    "command exits, use 'qiime tools extract'.",
                cls=ToolCommand)
-@click.argument('visualization-path', metavar='VISUALIZATION',
+@click.argument('result-path', metavar='RESULT',
                 type=click.Path(file_okay=True, dir_okay=False, readable=True))
-@click.option('--index-extension', required=False, default='html',
-              help='The extension of the index file that should be opened. '
-                   '[default: html]')
-def view(visualization_path, index_extension):
+@click.option('--port', required=False, type=click.IntRange(1024, 65535),
+              default=get_free_port(),
```

Comment:
I think the default should be `None` here, and then when None, you call the above `get_free_port()`, there will also be a race condition technically, as I think the port is released as soon as you close the socket, but I'm not sure if there's any way around that (i.e. can you recycle a socket with HTTPHandler? probably not).